### ISSUE TITLE
Feature request: Add early stopping mechanism to slim.evaluation_loop

### ISSUE THREAD

**Issue by yuvval** [tensorflow/tensorflow#7467]
Would it be possible to add early stopping mechanism to slim.evaluation_loop?

**Comment by aselle** [comment]
@sguadda, @nathansilberman  could you comment? Could this be contributed by external users? Is there a way to do it already?

**Comment by vg-pulsepoint** [comment]
+1 <bump>

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by scotthuang1989** [comment]
I think add support for hooks like 

> tf.train.MonitoredTrainingSession

would be a generic way to support this kind of requirements.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sguada: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sguada: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sguada: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sguada: It has been 16 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by sguada** [comment]
Closing until someone wants to contribute it.
